lib/utils/deprecation/deprecator.rb

```diff
@@ -0,0 +1,109 @@
+
+require 'utils/deprecation/config_file'
+require 'inspec/log'
+
+module Inspec
+  module Deprecation
+    class Deprecator
+      attr_reader :config, :groups
+
+      # This is used only in functional testing
+      def self.class_test_cfg_io(io)
+        @@test_cfg_io = io # rubocop: disable Style/ClassVars
+      end
+
+      def initialize(opts = {})
+        @@test_cfg_io ||= nil # rubocop: disable Style/ClassVars
+        @config = Inspec::Deprecation::ConfigFile.new(opts[:config_io] || @@test_cfg_io)
+        @groups = @config.groups
+      end
+
+      def handle_deprecation(group_name, message, opts = {})
+        group = groups[group_name.to_sym] # Will be nil if unrecognized group
```

Comment:
Throughout, the code has awkward conditionals because it cannot rely on having a GroupEntry record.  How could that be handled better?

Reply:
I've modified the code to create a GroupEntry when no group is found in the config file.